lib/jxl/splines.cc

```diff
@@ -536,8 +586,13 @@ Status Splines::InitializeDrawCache(const size_t image_xsize,
       continue;
     }
     HWY_DYNAMIC_DISPATCH(SegmentsFromPoints)
-    (spline, points_to_draw, arc_length, segments_, segments_by_y);
+    (spline, points_to_draw, arc_length, segments_, segments_by_y, &px_limit);
+    if (px_limit == 0) {
+      // Drop the splines that does not fit the budget.
+      break;
```

Comment:
Would it make sense to return an error instead?

Reply:
Going to revert to initial impl, as agreed.